File: Training/Classic Puzzle - Easy/Ascii_Art.py
"""
 @file Ascii_Art.py
 @author Ladjouzi Rachid (RashidLadj@gmail.com)
 @brief 
 @version 0.1
 @date 2021-10-25
 
 @copyright Copyright (c) 2021
"""

length_ = int(input())
heigth_ = int(input())
input_ = input()

alphabet = [input() for i in range(heigth_)]

for x in alphabet:
    tempLigne = []
    for actual_ in input_:
        if actual_ >= 'a' and actual_ <= 'z':
            idx = ord(actual_) - ord('a')
        elif actual_ >= 'A' and actual_ <= 'Z':
            idx = ord(actual_) - ord('A')
        else:
            idx = ord('z') - ord('a') + 1
        tempLigne.append(x[idx * length_: idx * length_ + length_])

    print("".join(tempLigne))

File: Training/Classic Puzzle - Easy/Ascii_Art.cpp

/**
 * @file Ascii_Art.cpp
 * @author Ladjouzi Rachid (RashidLadj@gmail.com)
 * @brief 
 * @version 0.1
 * @date 2021-10-26
 * 
 * @copyright Copyright (c) 2021
 */

#include <iostream>
#include <vector>

using namespace std;

int main(){
    int Length_, Heigth_;   cin >> Length_ >> Heigth_;  cin.ignore();
    string Input_;          getline(cin, Input_);
    
    vector<string> artASCII;
    for(int i = 0; i < Heigth_; i++){
        string row;        getline(cin, row);
        artASCII.push_back(row);
    }

    for (auto &row : artASCII){
        int idx;
        for (auto &actual_: Input_){
            if (actual_ >= 'a' and actual_ <= 'z')
                idx = int(actual_) - int('a');
            else if (actual_ >= 'A' and actual_ <= 'Z')
                idx = int(actual_) - int('A');
            else
                idx = int('z') - int('a') + 1;
        
            cout << row.substr(idx * Length_, Length_);
        }
        cout << endl;
    }

}